src/main/java/duke/utils/TaskList.java

```diff
@@ -0,0 +1,83 @@
+package duke.utils;
+
+import duke.task.Deadlines;
+import duke.task.Events;
+import duke.task.Task;
+import duke.task.Todos;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskList {
+
+    private List<Task> tasks = new ArrayList<>();
+
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public TaskList() { }
+
+    public List<Task> getList() {
+        return tasks;
+    }
+
+    @Override
+    public String toString() {
+        if (tasks.size() == 0) {
+            return "No scheduled task yet";
+        }
+        String list = "1." + tasks.get(0);
+        for (int i = 2; i < tasks.size() + 1; i++) {
+            list = list + "\n" + i + "." + tasks.get(i - 1);
+        }
+        return list;
+    }
+
+    public int size() {
+        return tasks.size();
+    }
+
+    public void markTaskAsDone(int taskId) {
+        tasks.get(taskId).markAsDone();
+    }
+
+    public void removeTask(int taskId) {
+        tasks.remove(taskId);
+    }
+
+    public boolean addToList(String item, String type) {
+        Task newTask;
+        if (type.equals("todo")) {
+            newTask = new Todos(item);
+        } else if (type.equals("deadline")) {
+            String[] tokens = item.split("/", 2);
```

Comment:
Might want to consider including type.equals("event") here as well so that Line 55 to 65 will not be repeated. After which do a if-else to decide whether to create an Event or a Deadline object.